packages/svelte-urql/src/queryStore.ts

```diff
@@ -0,0 +1,85 @@
+import { pipe, concatMap, subscribe, filter } from 'wonka';
+import type { OperationContext, OperationResult, Client } from '@urql/core';
+import { derived, writable } from 'svelte/store';
+import type { DocumentNode } from 'graphql';
+import type { TypedDocumentNode } from '@graphql-typed-document-node/core';
+import { createRequest } from '@urql/core';
+import type { AnnotatedOperationResult, UrqlStore, Pausable } from './common';
+import {
+  defaultBaseResult,
+  createPausable,
+  fetchProcess,
+  fromStore,
+} from './common';
+
+/**
+ * Creates a Svelte store for an [Urql query](https://formidable.com/open-source/urql/docs/api/core/#clientexecutequery) using [Wonka](https://wonka.kitten.sh/)
+ */
+export function queryStore<Data, Variables extends object = {}>(props: {
+  /** an [Urql client](https://formidable.com/open-source/urql/docs/api/core/#client)  */
+  client: Client;
```

Comment:
Why are we passing in a client manually rather than using it from context? Is this an ssr thing or?

Reply:
this is a big reason for the update (RFC https://github.com/FormidableLabs/urql/issues/2369)

In react, this "manual" pattern would be annoying.  In Svelte, I believe this is the preferable approach.

You may be looking for a more thorough answer... and I will be putting together some examples/docs for this to show three patterns:

1. simple case can just use the an `UrqlStore` directly
2. component-driven approach can use svelte context
3. separate-logic approach can use stores and doesn't need svelte context.
4. combine any of the three

Reply:
sorry for chiming in late, I'm still reading this as I find time :v:
I still find it odd to ignore Svelte's context entirely. I get the gap here and the gap context creates. However, I'd find it good if we had a way to continue providing a common context key that's in a way "reserved" for us, since this isn't identity based.

So, I suppose we could:
- Keep `getContextClient` & `setContextClient` intact (with an appropriate rename, not necessarily this one)
- Optionally and additionally: continue supporting context when no `client` is passed with new primitives
- or: export a shared context key that we recommend people to use when providing the `client` via Svelte's context

This way we don't require this but we also don't start ignoring that it's an option for people to use

Reply:
i really appreciate the feedback.

i can see instances where context could be a convenience.  i can also see why it might be unnerving to remove it (especially coming from a React perspective).  i can update it to keep context if you like, but every time i think about it with svelte, I keep coming back to the belief that it is better without context. quotes from a great article on svelte context:

> svelte context is commonly misunderstood [and people] end up reaching for it in many cases where other approaches are simpler.[instead] define a store in another Javascript file, and just import it everywhere that it’s needed.
https://imfeld.dev/writing/svelte_context

as the author points out, it is easy in svelte to use stores instead of context, so the benefit of using context is rather limited.

it might also be thought of as an [inheritance vs composition issue](https://en.wikipedia.org/wiki/Composition_over_inheritance).  svelte context behaves like inheritance and can easily become strange / messy. by removing the context approach, users are spared from finding themselves in an inheritance trap, and the urql binding is easier to learn, understand, and maintain.

if you still think this is needed, i should be able to bring it back in as you propose.  another idea is to see if we release v2 alpha and see if users want to get context back?

Reply:
>  define a store in another Javascript file, and just import it everywhere that it’s needed.

I can certainly agree with that and see how that saves a lot of trouble in Svelte. I'm not opposed to not having it be the default. However, I do think the point stands that it's wise to keep the `contextKey` reserved to us for future use. Hence, we could think of placing stores in that context, or otherwise create readables from the context value. It doesn't really matter, but I think we should continue to reserve that key at the very least since it's a constant literal 🤔

Reply:
I added an update for this.  is [my change here](packages/svelte-urql/src/context.ts) what you're recommending? (also added to exports)